pkg/apis/performanceprofile/v2/performanceprofile_validation.go

```diff
@@ -147,16 +147,16 @@ func (r *PerformanceProfile) validateCPUs() field.ErrorList {
 				}
 
 				if overlap := components.Intersect(cpuLists.GetIsolated(), cpuLists.GetReserved()); len(overlap) != 0 {
-					allErrs = append(allErrs, field.Invalid(field.NewPath("spec.cpu"), r.Spec.CPU, fmt.Sprintf("reserved and isolated cpus overlap: %v", overlap)))
+					allErrs = append(allErrs, field.Forbidden(field.NewPath("spec.cpu"), fmt.Sprintf("reserved and isolated cpus overlap: %v", overlap)))
```

Comment:
what if we call the `String()` methods explicitely? (adding them if needed, or string helpers)
```
allErrs = append(allErrs, field.Forbidden(field.NewPath("spec.cpu"), r.Spec.CPU.String(), fmt.Sprintf("reserved and isolated cpus overlap: %v", overlap)))
```

Reply:
You are talking about cpuset of kubelet. This is spec.cpu of API and has no String method. In fact `type CPUSet string`

It could be valid like this:
`allErrs = append(allErrs, field.Invalid(field.NewPath("spec.cpu"), string(*r.Spec.CPU.Reserved), fmt.Sprintf("reserved and isolated cpus overlap: %v", overlap)))`

But in any case, invalid would be the intersection, not the Reserved cpu, for that I've changed it to Forbidden.

Reply:
I'm also wondering if the field should actually be forbidden (why?) rather than invalid. No strong feelings so far.

Reply:
I don't think it's _necessary_. I do however think that keeping the context here will help and improve the UX, locally for this issue and in general.
Finding a good string representation for the CPU object can be nontrivial, so we can do in a later PR.

Reply:
Then if you agree we can move forward with this one and open other one to format the output of all invalid errors of this API
Also another thing because I've changed to forbidden its you don't have to add the arg "r.Spec.CPU"

Reply:
sounds good.